inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/convert_mul_or_add_finally.cpp

```diff
@@ -195,14 +195,14 @@ ngraph::matcher_pass_callback get_callback() {
                 auto biases_in = ngraph::op::util::normalize_constant(const_node, output_shape);
                 if (is_dequantization) {
                     const ngraph::Shape data_shape = data_node.get_shape();
-                    ngraph::Shape broadcasted_shape = std::vector<size_t>(data_shape.size(), 1ul);
+                    ngraph::Shape broadcasted_shape = weights_in->get_shape();
                     broadcasted_shape[1] = data_shape[1];
-
                     weights_in = ngraph::op::util::broadcastTo(weights_in, broadcasted_shape);
                     biases_in = ngraph::op::util::broadcastTo(biases_in, broadcasted_shape);
                 }
```

Comment:
@eshoguli could you explain the logic under `is_dequantization` condition?

Reply:
If operation is `Multiply` or `Add` then legacy `ScaleShift` have to be created (not `Power`, not `Eltwise`). This requirement was created by plugins in CNNNetwork time (CPU still requires CNNNetwork instance). In that time plugins were optimized to have `ScaleShift` layers as dequantization layers and were ready to fuse them.  
As result for 4D & 5D input tensors (if quantization is done per second dimension - `ScaleShift` limitation) we should create `ScaleShift` with broadcasted weights & biases.  
My suggestion is: to follow to this logic, while CPU uses legacy transformations. In the next release we remove that.

Reply:
I made changes, but I'm not sure that this condition is not too broad now. I will add tests when you accept the idea.

The related model didn't actually work, but with a completely different error:
Unsupported input sizes for Quantize layer with name

After merging the latest master (this pr'a: https://github.com/openvinotoolkit/openvino/pull/4460) the model started to infer correctly via benchmark_app

@eshoguli

Reply:
I have adjusted existing tests to match the new version, please review @eshoguli.

Please check mainly that the condition is not too general now, because I have no idea how to check it